.github/workflows/guix-build.yml

```diff
@@ -46,16 +46,21 @@ jobs:
           cache-from: type=gha
           cache-to: type=gha,mode=max
 
-      - name: Restore Guix cache and depends
+      - name: Cache Guix and depends
         id: guix-cache-restore
-        uses: actions/cache/restore@v3
+        uses: actions/cache@v3
         with:
           path: |
             ${{ github.workspace }}/.cache
             ${{ github.workspace }}/dash/depends/built
             ${{ github.workspace }}/dash/depends/sources
             ${{ github.workspace }}/dash/depends/work
-          key: ${{ runner.os }}-guix
+          key: ${{ runner.os }}-guix-${{ github.event.pull_request.head.sha }}
+          restore-keys: |
+            ${{ runner.os }}-guix-${{ github.event.pull_request.head.sha }}
+            ${{ runner.os }}-guix-
```

Comment:
is it typo, that `...-guix-` tag is over by a hyphen?

Reply:
wym?

Reply:
```
            ${{ runner.os }}-guix-${{ github.event.pull_request.head.sha }}
            ${{ runner.os }}-guix- <-------------------
```
it looks like copy-paste error. It seems as not full line, but cut in the middle.

If that's full line and it's a mask, why does it need a full key above then?

Reply:
no, it's intended, so the key is what it sets; which includes the commit id, and then the next ones are the "restore keys" so first we try the exact match, ie the one with the commit id, and then after that we try to match against one without the commit id.